internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,550 @@
+			}
+
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:    []string{string(network.ProvisioningStateUpdating)},
+				Target:     []string{string(network.ProvisioningStateSucceeded)},
+				Refresh:    r.provisioningStateRefreshFunc(ctx, id),
+				MinTimeout: 1 * time.Minute,
+				Timeout:    time.Until(timeout),
+			}
+			if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for provisioning state of %s: %+v", id, err)
+			}
+
+			desiredState := network.CommissionedStateProvisioned
+			if model.CommissioningEnabled {
+				if model.InternetAdvertisingDisabled {
+					desiredState = network.CommissionedStateCommissionedNoInternetAdvertise
+				} else {
+					desiredState = network.CommissionedStateCommissioned
+				}
+			}
+
+			if err := r.updateCommissionedState(ctx, id, desiredState); err != nil {
+				return err
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r CustomIpPrefixResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 17 * time.Hour,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			r.client = metadata.Client.Network.CustomIPPrefixesClient
+
+			id, err := parse.CustomIpPrefixID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Info("Decoding state...")
+			var state CustomIpPrefixModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			desiredState := network.CommissionedStateProvisioned
+			if state.CommissioningEnabled {
+				if state.InternetAdvertisingDisabled {
+					desiredState = network.CommissionedStateCommissionedNoInternetAdvertise
+				} else {
+					desiredState = network.CommissionedStateCommissioned
+				}
+			}
+
+			if err := r.updateCommissionedState(ctx, *id, desiredState); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomIpPrefixResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			r.client = metadata.Client.Network.CustomIPPrefixesClient
+
+			id, err := parse.CustomIpPrefixID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := r.client.Get(ctx, id.ResourceGroup, id.CustomIpPrefixeName, "")
+			if err != nil {
+				if utils.ResponseWasNotFound(existing.Response) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			model := CustomIpPrefixModel{
+				Name:              id.CustomIpPrefixeName,
+				ResourceGroupName: id.ResourceGroup,
+				Location:          location.StateFunc(existing.Location),
+				Tags:              tags.ToTypedObject(existing.Tags),
+			}
+
+			if existing.Zones != nil {
+				model.Zones = *existing.Zones
+			}
+
+			if props := existing.CustomIPPrefixPropertiesFormat; props != nil {
+				if props.Cidr != nil {
+					model.CIDR = *props.Cidr
+				}
```

Comment:
Can use this for simplicity?
```suggestion
				model.CIDR = pointer.From(props.Cidr)
```